src/util.rs

```diff
@@ -81,44 +81,45 @@ pub(crate) mod testing {
         };
     }
 
-    pub fn test_roundtrip_with_args<T>(
+    pub fn test_roundtrip_with_args<'a, T>(
         bin: &[u8],
         obj: T,
-        read_args: <T as binrw::BinRead>::Args,
-        write_args: <T as binrw::BinWrite>::Args,
+        read_args: <T as binrw::BinRead>::Args<'a>,
+        write_args: <T as binrw::BinWrite>::Args<'a>,
     ) where
+        <T as binrw::BinRead>::Args<'a>: Clone,
+        <T as binrw::BinWrite>::Args<'a>: Clone,
         T: BinRead + BinWrite + PartialEq + core::fmt::Debug + ReadEndian + WriteEndian,
```

Comment:
This looks a little weird to me, if we require the `Args` to satisfy `Clone` anyways, can we avoid `'a` on the function somehow?

Reply:
Mh, I'm unsure how to achieve that. When just removing the `'a` from the function signature, it yields this:

```
error[E0261]: use of undeclared lifetime name `'a`
  --> src/util.rs:90:37
   |
90 |         <T as binrw::BinRead>::Args<'a>: Clone, 
   |                                     ^^ undeclared lifetime
   |
   = note: for more information on higher-ranked polymorphism, visit https://doc.rust-lang.org/nomicon/hr
tb.html
help: consider making the bound lifetime-generic with a new `'a` lifetime
   |
90 |         for<'a> <T as binrw::BinRead>::Args<'a>: Clone,
   |         +++++++
help: consider introducing lifetime `'a` here
   |
84 |     pub fn test_roundtrip_with_args<'a, T>(
   |                                     +++


```

Setting the lifetime seems to be required for `Args` and leaving it empty as in `'_` doesn't work either.

Reply:
Yeah, thats weird. I'm not deep enough into Rust to understand this either. I guess we can ignore it until someone shows up with (or I acquire it) the skill to look into this.